webrender/src/device/gfx/device.rs

```diff
@@ -1483,13 +1479,28 @@ impl<B: hal::Backend> Device<B> {
                 ],
             );
 
-            let (desc_set, need_alloc) = match self.per_draw_descriptor_bindings[self.next_id].entry(bound_resources) {
-                Entry::Occupied(o) => {
-                    (self.descriptor_pools_per_draw[self.next_id].get_set_at_location(shader_group, *o.get()), false)
+            // Check if we have a free to use descriptor set for these bindings, or we need to create a new one
+            let (desc_set, need_bind) = match self.per_draw_descriptor_bindings.entry(bound_resources) {
+                Entry::Occupied(mut info_vec) => {
+                    let has_free_set = info_vec.get().into_iter().any(|i| !i.still_in_flight(frame_id, frame_count));
```

Comment:
This shouldn't be necessary. We can re-use the descriptors that are in flight if we are not mutating them. So we don't need a vector of descriptor sets per `bound_resources`, we just need one. If we look it up and we find it, all we need to do is to update the `bound_in_frame` to the latest.